Lesson1HomeTask/task1/main.cpp

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <algorithm>
+
+using my_vec_t = std::vector<int>;
+
+my_vec_t my_merge1 (const my_vec_t& v1, const my_vec_t& v2)
+{
+    my_vec_t result {v1};
+    const auto  result_size = v1.size() + v2.size();
+    result.resize(result_size);
+
+    auto it = std::next (result.begin(), v1.size());
+
+    std::copy (v2.begin(), v2.end(), it);
+    return result;
+}
+
+my_vec_t my_merge2 (const my_vec_t& v1, const my_vec_t& v2)
+{
+    my_vec_t result {v1};
+    for (const auto& el : v2)
+        result.push_back (el);
+    return result;
+}
+
+my_vec_t my_merge3 (const my_vec_t& v1, const my_vec_t& v2)
+{
+    my_vec_t v1_copy = v1;
+    my_vec_t v2_copy = v2;
+
+    std::sort (v1_copy.begin(), v1_copy.end());
+    std::sort (v2_copy.begin(), v2_copy.end());
+
+    my_vec_t result;
+
+    result.resize (v1.size() + v2.size());
+
+    std::merge (v1_copy.begin(), v1_copy.end(),
+                v2_copy.begin(), v2_copy.end(),
+                result.begin());
+
+    return result;
+}
+
+
+void print_vector (const my_vec_t& vec)
+{
+    for (const auto& el : vec)
+        std::cout << el << "  ";
+}
+
+void run_test (const my_vec_t& v1, const my_vec_t& v2, my_vec_t (*merge_func)(const my_vec_t& v1, const my_vec_t& v2))
```

Comment:
If you want, you can try to use gtest here (it is optionally, but it will be required in next lessons). At least, you can try static asserts. 
http://www.cplusplus.com/reference/cassert/assert/

Reply:
Ok, I will next time)